in_toto/verifylib.py

```diff
@@ -1048,8 +1050,16 @@ def verify_disallow_rule(rule, source_artifacts_queue):
       source_artifacts_queue, rule_data["pattern"])
 
   if len(matched_artifacts):
-    raise RuleVerificationError("Rule '{0}' failed, pattern matched disallowed"
-        " artifacts: '{1}' ".format(" ".join(rule), matched_artifacts))
+    rule_traceback = "Initial state of queues:\n"
+    rule_traceback += pprint.pformat(RULE_TRACE["trace"][0])
+    for num in range(1,len(RULE_TRACE["trace"])):
+      curr_rnd = RULE_TRACE["trace"][num]
+      rule_name = curr_rnd["rule"]
+      queues = {k: curr_rnd[k] for k in ('artifacts', 'materials', 'products')}
+      q_prompt = "\nQueues after processing of rule '{0}':\n".format(rule_name)
+      rule_traceback += q_prompt + pprint.pformat(queues)
```

Comment:
Oh, I must have been unclear. I was thinking of a helper function for L1053-L1060 so as to not pollute `verify_disallow_rule` with all that dict parsing and string constructing code.

The error message that you moved out now could IMHO still be left here, and suffixed by the return value from said helper function,  which might look something liked this:
```python
def _get_rule_traceback():
    trace_str = "Trace of {} rules for item '{}':\n".format(
        RULE_TRACE["type"], RULE_TRACE["name"])

    trace_str += "Initial state of queues:\n"
    trace_str += pprint.pformat(RULE_TRACE["trace"].pop(0)) + "\n"

    for trace_line in RULE_TRACE["trace"]:
      trace_str += "Queues after processing of rule '{0}':\n".format(
          trace_line.pop("rule"))
      trace_str += pprint.pformat(trace_line) + "\n"
```

What do you think?

Reply:
I agree with your suggestions. It was my mistake for misunderstanding your intentions. I will be sure to update the functionality of the helper function to reflect this.